Jenkinsfile

```diff
@@ -150,7 +150,34 @@ pipeline {
               }
             }
           }
-        } 
+        }
+        stage('TPU VM') {
+          stages {
+            stage('Build Tensorflow TPU Image') {
+              options {
+                timeout(time: 120, unit: 'MINUTES')
```

Comment:
How long does this step take?

Reply:
I've never run it on Jenkins so I guess we don't know yet for sure. Building the tensorflow image takes considerable time though :(

Reply:
How long does building the TPU VM image takes? (without building TensorFlow)?

I am concern with adding an extra 2 hours to every build. Having this build complete in the shortest amount of time is very valuable for me to keep the maintenance cost low and iterate fast.

The TensorFlow Image probably doesn't need to be rebuilt every time. It should only be rebuilt when the base image or the TensorFlow image is updated. We could use a similar approach than what I use for building lightgbm and torch from source for the GPU image. I basically build the packages in parallel and the pipeline only rebuild the image if the base image or the package version is updated: 

- logic for parallel build: https://github.com/Kaggle/docker-python/blob/389cecd4683774c7ce33d39667dd788742e77816/Jenkinsfile#L24-L59
- logic for conditional build: https://github.com/Kaggle/docker-python/blob/389cecd4683774c7ce33d39667dd788742e77816/packages/build_package#L118-L131

Reply:
Thanks! I think I've adjusted it to fit this (merged it into the "build_packages" setup).